test/extended/cli/mustgather.go

```diff
@@ -173,81 +173,96 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
-				o.Expect(err).NotTo(o.HaveOccurred())
+					fi, err := file.Stat()
+					o.Expect(err).NotTo(o.HaveOccurred())
 
-				// it will happen that the audit files are sometimes empty, we can
-				// safely ignore these files since they don't provide valuable information
-				if fi.Size() == 0 {
-					return nil
-				}
+					// it will happen that the audit files are sometimes empty, we can
+					// safely ignore these files since they don't provide valuable information
+					// TODO this doesn't seem right.  It should be really unlikely, but we'll deal with later
+					if fi.Size() == 0 {
+						return nil
+					}
+
+					gzipReader, err := gzip.NewReader(file)
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					scanner := bufio.NewScanner(gzipReader)
+					for scanner.Scan() {
+						text := scanner.Text()
+						if !strings.HasSuffix(text, "}") {
+							continue // ignore truncated data
+						}
+						o.Expect(text).To(o.HavePrefix(`{"kind":"Event",`))
+						for _, token := range []string{"oauthaccesstokens", "oauthauthorizetokens", tokenName} {
+							o.Expect(text).NotTo(o.ContainSubstring(token))
+						}
+						readFile = true
+						eventsChecked++
+					}
+					// ignore this error as we usually fail to read the whole GZ file
+					// o.Expect(scanner.Err()).NotTo(o.HaveOccurred())
+					o.Expect(readFile).To(o.BeTrue())
 
-				gzipReader, err := gzip.NewReader(file)
+					return nil
+				})
 				o.Expect(err).NotTo(o.HaveOccurred())
 
-				scanner := bufio.NewScanner(gzipReader)
-				for scanner.Scan() {
-					text := scanner.Text()
-					if !strings.HasSuffix(text, "}") {
-						continue // ignore truncated data
-					}
-					o.Expect(text).To(o.HavePrefix(`{"kind":"Event",`))
-					for _, token := range []string{"oauthaccesstokens", "oauthauthorizetokens", tokenName} {
-						o.Expect(text).NotTo(o.ContainSubstring(token))
-					}
-					readFile = true
-					eventsChecked++
+				if eventsChecked <= expectedNumberOfAuditEntries {
+					lastErr = fmt.Errorf("expected %d audit events for %q, but only got %d", expectedNumberOfAuditEntries, auditDirectory, eventsChecked)
+					return false, nil
 				}
-				// ignore this error as we usually fail to read the whole GZ file
-				// o.Expect(scanner.Err()).NotTo(o.HaveOccurred())
-				o.Expect(readFile).To(o.BeTrue())
 
-				return nil
-			})
-			o.Expect(err).NotTo(o.HaveOccurred())
-
-			// On IBM ROKS, events will not be part of the output, since audit logs do not include
-			// control plane logs.
-			if e2e.TestContext.Provider != ibmcloud.ProviderName {
-				o.Expect(eventsChecked).To(o.BeNumerically(">", 1000))
+				// reset lastErr if we succeeded.
+				lastErr = nil
 			}
-		}
+
+			// if we get here, it means both directories checked out ok
+			return true, nil
+		})
+		o.Expect(lastErr).NotTo(o.HaveOccurred()) // print the last error first if we have one
+		o.Expect(err).NotTo(o.HaveOccurred())     // otherwise be sure we fail on the timeout if it happened
```

Comment:
"context expired" or whatever we get here is probably not going to be all that useful.  Can we include "no audit directories found" or some such in this error message?